Jobs/PrGate.yml

```diff
@@ -46,7 +47,7 @@ parameters:
 - name: linux_container_image
   displayName: Linux Container Image
   type: string
-  default: 'ghcr.io/tianocore/containers/fedora-35-build:5b8a008'
+  default: 'ghcr.io/tianocore/containers/fedora-37-build:f1c7a20'

```

Comment:
I ultimately want to get to the point where we track the exact container image (by SHA) in one place. Right now that is Version.njk.

My thought was:

1. Any pipeline that cares to use a specific version should pass it in as a parameter (Version.njk is one way to get the value and do that)
2. The default requires less maintenance over time

With that in mind, can this just be `"fedora-37-build:latest"` for now? As long as latest works as expected in the tianocore container repo (it does not always depending on how you manage images).

Reply:
My biggest concern with the latest is that if we support forked branches, they would keep getting the new container but with all the old build definitions. It's possible there could be incompatibilities there we would have to concern ourselves with like dropping tools that are no longer used, etc.

Reply:
Does any code depend on the default value right now? Ultimately, I think all template users should pass this value since they need to account for changes in the image. For example, this could be an empty default and we raise a pipeline error if it remains empty.

If the repo opts in to receive its "leaf" pipelines from file sync, they get it automatically (from Version.njk). Otherwise, they can pull Version.njk down in their pipeline run and extract the value.

Reply:
> Is pulling version.njk feasible for Azure dev-ops repos?

Sure. Here's a quick curl example:
```bash
curl -s https://raw.githubusercontent.com/microsoft/mu_devops/main/.sync/Version.njk | grep linux_build_container | cut -d '"' -f 2
```

What I'm proposing is that we do this one way. Either (1) require template users to specify what container image they want or (2) always specify the container image in the pipeline.

By this being a template parameter, that means (1) has been chosen. So, the complexity associated with testing that is inherited. (2) brings more complexity that I don't think is needed if (1) is supported. I think (1) makes sense because it allows individual repos to control their build environment.

Then, it becomes a testing problem. Which is an issue (#23). But, because this repo produces versioned release, dependent repos do not need to update what mu_devops they use until they can verify their chosen container image works with the mu_devops version they are updating to and that can be confirmed in their CI build process.

Reply:
Yeah, I suppose I see this as a convenience. For all the synced repos it's moot really since they get their top-level pipeline files from this repo anyways so the burden of testing really doesn't change one way or the other in my opinion. For all the non-synced repos this allows an easy way to just stay in lock-step with dev_ops. All such repos will need to manually roll their dev_ops version anyways, so whether they need to validate the container separately or not seems to make no difference testing-wise in my mind. This just provides an easy solution if they don't need container customizations.

Either way, I don't think there are any non-synced repros that use containers anyways, so i'm happy to just push this to the azure_pipelines/custom/* files using version.njk if that work for you.